api/placer.py

```diff
@@ -255,15 +306,27 @@ def check(self):
         self.zip.write(self.tempdir.name, self.dir)
 
     def process_file_field(self, field, info):
-        # Set the file's mtime & atime.
-        os.utime(field.path, (self.ziptime, self.ziptime))
-
-        # Place file into the zip folder we created before
-        self.zip.write(field.path, os.path.join(self.dir, field.filename))
+        # Should not be called with any files
+        raise Exception('Files must already be uploaded')
 
     def finalize(self):
+
+        # Write all files to zip
+        for path in os.listdir(self.folder):
+            p = os.path.join(self.folder, path)
+
+            # Set the file's mtime & atime.
+            os.utime(p, (self.ziptime, self.ziptime))
+
+            # Place file into the zip folder we created before
+            self.zip.write(p, os.path.join(self.dir, os.path.basename(path)))
+
         self.zip.close()
 
+        print self.folder
+        print 'SAFETY'
+        # shutil.rmtree(self.folder)
```

Comment:
Disabled for obvious reasons while developing. Can be turned on and debug statements removed.